pkg/controller/machine.go

```diff
@@ -479,7 +488,9 @@ func (c *Controller) ensureInstanceExistsForMachine(prov cloud.Provider, machine
 			return fmt.Errorf("invalid provider config: %v", err)
 		}
 		c.recorder.Event(machine, corev1.EventTypeNormal, "ValidationSucceeded", "Validation succeeded")
+		c.validationCacheMutex.Lock()
 		c.validationCache[cacheKey] = true
+		c.validationCacheMutex.Unlock()
```

Comment:
Why do we need a mutex here?
As a machine does not get processed concurrently i don't see a any benefit here.
Sure workers access the map concurrently but always with a different key

Reply:
I introduced it after my local test machine controller panicked with a concurrent write to the sync map

Reply:
I think that the golang runtime detects concurrent writes to the map and marks them as an error   .

Reply:
Good to know, though always it only does this for concurrent access on the same key